Makefile

```diff
@@ -218,6 +219,41 @@ build/compiled-by-previous-traceur.js: \
 debug: build/compiled-by-previous-traceur.js $(SRC)
 	./traceur --debug --out bin/traceur.js --sourcemap $(RUNTIME_SCRIPTS) $(TFLAGS) $(SRC)
 
+#
+# Rules to test traceur.js compiled through path that first compiles out modules then es6.
+# The regular compiler must be built before running these rules.
+#
+
+build/es6-no-modules/compiler.js: $(SRC_NODE) src/traceur-import.js
+	node_modules/traceur/traceur --out $@ --modules=inline  --referrer='traceur@0.0.0/build/es6-no-modules/' \
+	  --outputLanguage=es6 $(TFLAGS) src/traceur-import.js
+
+build/es6-no-modules/globalModules.js: $(RUNTIME_MODULES) $(POLYFILL_SRC) src/runtime/runtimeModules.js src/runtime/polyfillsModules.js
+	./traceur --out build/es6-no-modules/globalModules.js --modules=inline --outputLanguage=es6 -- src/runtime/globalModules.js
+
+build/es6-no-modules/runtime.js: $(RUNTIME_SRC) build/es6-no-modules/globalModules.js
+	./traceur --out $@ $(RUNTIME_SCRIPTS) $(TFLAGS)
+	cat build/es6-no-modules/globalModules.js >> $@
+
+# The complete traceur compiler in es6 but with modules compiled away.
+bin/traceur-es6-no-modules.js: build/es6-no-modules/runtime.js build/es6-no-modules/compiler.js
+	cat $^ > $@
```

Comment:
This makes sourcemaps fail.

Also this runtime contains System and a bunch of ES6 polyfills which we might not want.